src/transformers/cache_utils.py

```diff
@@ -777,7 +786,20 @@ def __init__(self, cache_config: CacheConfig) -> None:
 
     def _quantize(self, tensor, axis):
         scale, zeropoint = self.optimizer(tensor, self.qtype.bits, axis, self.q_group_size)
-        qtensor = AffineQuantizer.apply(tensor, self.qtype, axis, self.q_group_size, scale, zeropoint)
+        # We have two different API since in optimum-optimun, we don't use AffineQuantizer anymore
+        if is_optimum_quanto_available():
+            from optimum.quanto import QBitsTensor
+
+            qtensor = QBitsTensor.quantize(tensor, self.qtype, axis, self.q_group_size, scale, zeropoint)
+            return qtensor
+        elif is_quanto_available():
+            logger.warning_once(
+                "Importing from quanto will be deprecated in v4.47. Please install optimum-quanto instead `pip install optimum-quanto`"
+            )
+            from quanto import AffineQuantizer
+
+            qtensor = AffineQuantizer.apply(tensor, self.qtype, axis, self.q_group_size, scale, zeropoint)
+
```

Comment:
Does this sounds good @dacorvo ? I checked quantize_weights but they are BC between main and the latest version with args being removed. Also you use introduced `WeightQBitsTensor` recently in quantize_weights.

Reply:
This will break on next release: I am doing it right now so that at least you can use the latest API.

Reply:
Here the new release: https://github.com/huggingface/optimum-quanto/releases/tag/v0.2.5